src/macros.rs

```diff
@@ -267,3 +267,33 @@ macro_rules! define_casts {
         )+
     }
 }
+
+#[macro_export]
+macro_rules! cfg_feature_enabled {
+    ($name:tt) => (
+        {
+            #[cfg(target_feature = $name)] {
```

Comment:
Could this use `cfg!` to be a bit more compact?

Reply:
You mean `if cfg!(...) { ... }` ?

Reply:
Ah I was thinking like:

```rust
cfg!(target_feature = $name) || __unstable_detect_feature!($name)
```

and then the `__unstable_detect_feature` would do all sorts of macro craziness

Reply:
I could do it like that, but then the code is generated for both cases, both branches must type check, etc.

Reply:
That's true... 

I guess I'd just favor library readability over microseconds of compile time downstream

Reply:
That is probably a bad habit of mine. I use `#[cfg(...)]` when I want conditional compilation because that's what it does. 

I use `cfg!` to do unconditional compilation of all branches because that's what it does but... that's something I basically almost never want. If you give me a nice if-like feature for conditional compilation I'll use it every time (didn't you write a crate for that?), but until then, they mean different things.

Reply:
I also think that we should get rid of the `cfg!` habit when working with target features. By using `cfg!` one is basically inserting unwanted intrinsic calls in dead branches of the code. Sure LLVM should remove those, but call me paranoid if I don't want to trust what happens if LLVM actually sees one of those and then concludes that your whole function must be AVX2 enabled or similar.